x-pack/plugins/files/server/file_client/integration_tests/es_file_client.test.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Readable } from 'stream';
+import { loggingSystemMock } from '@kbn/core/server/mocks';
+import { TestEnvironmentUtils, setupIntegrationEnvironment } from '../../test_utils';
+import { createEsFileClient } from '../create_es_file_client';
+import { FileClient } from '../file_client';
+
+/**
+ * This file client is using Elasticsearch interfaces directly to manage files.
+ */
+describe('ES-index-backed file client', () => {
+  let esClient: TestEnvironmentUtils['esClient'];
+  let fileClient: FileClient;
+  let testHarness: TestEnvironmentUtils;
+
+  beforeAll(async () => {
+    testHarness = await setupIntegrationEnvironment();
+    ({ esClient } = testHarness);
+  });
+
+  beforeEach(() => {
+    fileClient = createEsFileClient({
+      blobStorageIndex: '.kibana-test-blob',
+      metadataIndex: '.kibana-test-metadata',
+      elasticsearchClient: esClient,
+      logger: loggingSystemMock.create().get(),
+    });
+  });
+
+  afterAll(async () => {
+    await testHarness.cleanupAfterAll();
+  });
+
+  test('create a new file', async () => {
+    const file = await fileClient.create({
+      id: '123',
+      metadata: {
+        Status: 'AWAITING_UPLOAD',
+        created: new Date().toISOString(),
+        Updated: new Date().toISOString(),
+        name: 'cool name',
+      },
+    });
+    expect(file).toEqual(
+      expect.objectContaining({
+        id: '123',
+        metadata: {
+          FileKind: 'none',
+          Status: 'AWAITING_UPLOAD',
+          Updated: expect.any(String),
+          created: expect.any(String),
+          name: 'cool name',
+        },
+      })
+    );
+    await fileClient.delete({ id: file.id, hasContent: false });
+  });
+
+  test('uploads and downloads file content', async () => {
```

Comment:
Would be great to add one more test case here: which uploads say 11 MiB file, and then checks all the chunks that they have 4 MiB, 4 MiB, and 3 MiB of data each with expected metadata and data.

Reply:
Would be also good to test edge cases:

- 0 bytes file
- exactly 4 MiB file
- exactly 8 MiB file